File: java/basics/1672.richest-customer-wealth.java
package basics;

/*
 * @lc app=leetcode id=1672 lang=java
 *
 * [1672] Richest Customer Wealth
 */

// @lc code=start

import java.util.stream.IntStream;

class Solution {
    public int maximumWealth(int[][] accounts) {
        int largest = 0;
        for (int[] account : accounts) {
            int sum = IntStream.of(account).sum();
            if (sum > largest) {
                largest = sum;
            }
        }
        return largest;
    }
}
// @lc code=end

File: dart/basics/1672.richest-customer-wealth.dart
/*
 * @lc app=leetcode id=1672 lang=dart
 *
 * [1672] Richest Customer Wealth
 */

// @lc code=start
class Solution {
  int maximumWealth(List<List<int>> accounts) {
    return accounts.fold(0, (int previousWealth, List<int> customerAccounts) {
      int customerWealth = customerAccounts.reduce((int a, int b) => a + b);
      return previousWealth > customerWealth ? previousWealth : customerWealth;
    });
  }

  /* Smelly 1liner
  import 'dart:math';
  int maximumWealth(List<List<int>> accounts) => accounts.fold(0, (int previousWealth, List<int> customerAccounts) => max(previousWealth, customerAccounts.reduce((int a, int b) => a + b)));
  */

  /* original solution
  int maximumWealth(List<List<int>> accounts) {
    List<int> customerWealth = [];
    for (List<int> customerAccounts in accounts) {
      int summedWealth = customerAccounts.reduce((int a, int b) => a + b);
      customerWealth.add(summedWealth);
    }
    int maxWealth = customerWealth.reduce((int a, int b) => a > b ? a : b);
    return maxWealth;
  }
  */
}
// @lc code=end
